### PR TITLE
feat(DDM): add ddm account resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add ddm account resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add ddm account resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ddm/' TESTARGS='-run TestAccDdmAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ddm/ -v -run TestAccDdmAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccDdmAccount_basic
=== PAUSE TestAccDdmAccount_basic
=== CONT  TestAccDdmAccount_basic
--- PASS: TestAccDdmAccount_basic (471.74s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ddm       471.788s
```

### CODE REVIEW

File: docs/resources/ddm_account.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the ID of a DDM instance.
+
```
Comment (deer-hang) [inline docs/resources/ddm_account.md]: Please remove the blank line and check the below.

File: docs/resources/ddm_account.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 30 minutes.
+* `update` - Default is 10 minutes.
+* `delete` - Default is 10 minutes.
+
+## Import
+
+The ddm account can be imported using the `<instance_id>/<account_name>`, e.g.
```
Comment (deer-hang) [inline docs/resources/ddm_account.md]: The field `account_name` does not appear in the document, please change it to `name`.
Please add `shell` on the follow example.

File: huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	}
+	for _, account := range accounts.([]interface{}) {
+		name := utils.PathSearch("name", account, nil)
+		if accountName != name {
+			continue
+		}
+		return account, nil
+	}
+
+	return nil, fmt.Errorf("the instance %s has no account", instanceID)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go]: Make the error message clear.
`return nil, fmt.Errorf("the instance %s has no account %s", instanceID, accountName)`

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+			}
+			if apiError.ErrorCode == "DBS.200019" {
+				err = waitForInstanceRunning(ctx, d, cfg, instanceID,
+					[]string{"DROP_DATABASE", "CREATE_DATABASE", "BACKUP"}, schema.TimeoutCreate)
+				if err != nil {
+					return diag.FromErr(err)
+				}
+				continue
+			}
+		}
```
Comment (deer-hang) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go:165]: The code is used in many places, extract a common method.

File: huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+			{
+				Config: testDdmAccount_basic(instanceName, name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttr(rName, "permissions.0", "SELECT"),
+				),
+			},
+			{
+				Config: testDdmAccount_basic_update(instanceName, name),
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go:102]: please add a test when update password

File: docs/resources/ddm_account.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/ddm_account.md]: done

File: docs/resources/ddm_account.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/ddm_account.md]: done

File: huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go]: done

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go:165]: done

File: huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/ddm/resource_huaweicloud_ddm_account_test.go:102]: the password cannot get by the API, so it cannot be test

File: docs/resources/ddm_account.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 30 minutes.
+* `update` - Default is 10 minutes.
+* `delete` - Default is 10 minutes.
+
+## Import
+
+The ddm account can be imported using the instance ID and account name separated by a slash, e.g.:
```
Comment (ShiChangkuo) [inline docs/resources/ddm_account.md]:
```suggestion
The DDM account can be imported using the instance ID and account name separated by a slash, e.g.:
```

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go [hunk 7]
```diff
@@ -0,0 +1,518 @@
...
+
+	var createAccountResp *http.Response
+	for {
+		createAccountResp, err = createAccountClient.Request("POST", createAccountPath, &createAccountOpt)
+		if err == nil {
+			break
+		}
+		err = handleOperationError(ctx, d, cfg, err, instanceID, schema.TimeoutCreate)
+		if err != nil {
+			return diag.Errorf("error creating DdmAccount: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go]:
```suggestion
			return diag.Errorf("error creating DDM account: %s", err)
```
please update them in all files

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go [hunk 8]
```diff
@@ -0,0 +1,518 @@
...
+			d.Set("name", name),
+			d.Set("status", utils.PathSearch("status", account, nil)),
+			d.Set("permissions", utils.PathSearch("base_authority", account, nil)),
+			d.Set("description", utils.PathSearch("description", account, nil)),
+			d.Set("schemas", flattenGetAccountResponseBodyDatabase(account)),
+		)
+		break
+	}
+
+	return diag.FromErr(mErr.ErrorOrNil())
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go]: if the user is not found, we should remove the resource in state

Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go]: done

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go (hunk 7 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_account.go]: done

File: docs/resources/ddm_account.md (hunk 6 above)
Comment (houpeng80) [reply docs/resources/ddm_account.md]: done

**Pull request merged**